src/rust/src/zcashd_orchard.rs

```diff
@@ -33,3 +34,44 @@ impl Ord for OrderedAddress {
         (&self.to_raw_address_bytes()).cmp(&other.to_raw_address_bytes())
     }
 }
+
+/// Internal newtype wrapper that allows us to use nullifiers as
+/// BTreeMap keys.
+#[derive(Clone, Copy)]
+#[repr(transparent)]
+pub(crate) struct OrderedNullifier(Nullifier);
```

Comment:
Didn't we just add `PartialOrd, Ord` impls to `Nullifier`? So this type is unnecessary now.